pkg/kubelet/kubelet.go

```diff
@@ -2431,6 +2437,47 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}
 }
 
+// CheckpointContainer tries to checkpoint a container. The parameters are used to
+// look up the specified container. If the container specified by the given parameters
+// cannot be found an error is returned. If the container is found the container
+// engine will be asked to checkpoint the given container into the kubelet's default
+// checkpoint directory.
+func (kl *Kubelet) CheckpointContainer(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+) error {
+	container, err := kl.findContainer(podFullName, podUID, containerName)
+	if err != nil {
+		return err
+	}
+	if container == nil {
+		return fmt.Errorf("container %v not found", containerName)
+	}
+
+	if strings.HasPrefix(options.Location, "/") {
+		return fmt.Errorf("archive_location cannot start with '/' but %s was provided", options.Location)
+	}
+
+	if !strings.Contains(options.Location, "/") && options.Location != "" {
```

Comment:
Missing a test case that covers the negative case of this path - not quite sure what the intent of the behavior is though.

Reply:
Added an additional test case. That code path will be the main code path. Saving the checkpoint image in the local registry for easy distribution to other registries.